### DISCUSSION TITLE
Feature request: Increase 128Mb deployment file size limit

### DISCUSSION THREAD

**Discussion by pauldequant** [umbraco/Umbraco.Cloud.Issues#891] [Ideas]
### Issue description

When using **CI/CD**, there is currently a website deployment file size limitation of **128Mb**. This restricts our ability to deploy our site in it's entirety. I would like to suggest that this limit is either removed (ideally) or at the very least increased to **1024Mb** or the highest possible value technically feasible. 

In addition, limitations such as this should be disclosed to customers before Cloud Plans are purchased, as this could be a deciding factor when migrating websites to the platform. There is no point purchasing any cloud plan, if you can't actually upload your website to it.

Thanks in advance 👍

**Comment by mikkelhm** [comment]
Hi @pauldequant 

Thanks for reaching out the the Cloud Team on this one.

I'll try to follow up a bit with some questions that I would like to get answered, to gain a little more knowledge about how you are utilizing the Cloud repositories, ill also add a little explanation about why the limit is there, and why we haven't raised it yet.

When you are doing a CICD deployment it will essentially end up updating the git repository that is behind your Umbraco Cloud environment. That means that all the files that are part of the package you are uploading will end up being part of that git repository.

By nature git repositories shouldn't contain large amounts of files or big files for that sake. Eventhough new mechanisms are available to handle large quantum of files and larger files, then the git repositories on Umbraco Cloud should still be kept as lean and clean as possible. 
The reason is that when we are doing any operations on Umbraco Cloud, we need to use the git repository. For instance when changing a plan or doing an upgrade, we are taking the current repository and doing changes to it. Having huge repositories will greatly slow any of those processes down, and make them more unstable. 
So in short for technical reasons we'd like those repositories to be as small as possible. When creating a new Umbraco Cloud repository the git repository is around 60KB.

You do need to add your own assets etc. to that repository, but ideally what is pushed to Cloud is just c# files and compiled javascript+css files. 

For the above reason, I'm interested in knowing why the packages reaches more than 128MB, is there something we can do or is missing, to make sure that it doesn't reach those sizes?

**Comment by pauldequant** [comment]
Hi @mikkelhm,

Thanks for following this up. So the website in question contains a lot of html interactive tools. So there are lot's of files outside of the Umbraco CMS - HTML, CSS, FONTS, Images etc. Without these files, our website footprint is about 80-90Mb, but with them you're looking at ~280-300Mb. We are reviewing what we upload in the hopes of reducing that, but I think it's unlikely we'd get it to less than 128Mb and even then, if our client wanted to upload more tools in the future, then we'd be in this position all over again.

If we could FTP them that would be something, we could just exclude the tools entirely, but I don't think that's an option you can provide.

**Comment by c9mb** [comment]
Yes, perhaps an sftp connection service is a required feature?

Obviously Azure WebApps has path-mapping options in its configuration, with the ability to create virtual directories and even mount additional storage.

e.g. Add SFTP --->  Username = {sitename}-username   Password = ******   Root-Path = .../myfiles

.../myfiles is then added as a virtual path to the webapp, which can be excluded from the git repository, and the SFTP service is used to upload files to it.
